lib/models/category.dart

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'package:flutter/material.dart';
+
+/// A data object representing a category.
+class Category {
+  Category({
+    required this.title,
+    required this.icon,
+    required this.backgroundImage,
+  });
+
+  final String title;
+  final IconData icon;
+  final String backgroundImage;
+}
+
+enum SampleCategory {
+  all('All'),
+  analysis('Analysis'),
+  // augmentedReality('Augmented Reality'),
+  cloudAndPortal('Cloud and Portal'),
+  editAndManageData('Edit and Manage Data'),
+  layers('Layers'),
+  maps('Maps'),
+  routingAndLogistics('Routing and Logistics'),
+  // scenes('Scenes'),
+  searchAndQuery('Search and Query'),
+  // utilityNetworks('Utility Networks'),
+  visualization('Visualization');
+  // favorites('Favorites');
+  const SampleCategory(this.title);
+  final String title;
```

Comment:
I think some code can be simplified here. We could add "icon" and "backgroundImage" to `enum SampleCategory`, and then get rid of `class Category` entirely.

```dart
enum SampleCategory {
  all('All', Icons.apps, 'assets/category_images/all_background.webp');
  // ... all the other values ...
  const SampleCategory(this.title, this.icon, this.backgroundImage);
  final String title;
  final IconData icon;
  final String backgroundImage;
}
```

Then, we could get rid of the `sampleCategories` list and just use `SampleCategory.values` in its place.

Reply:
Also that way we can stop using `import ... as arcgis_category;`, because there won't be a collision with the name of the `Category` class.